pkg/cmd/server/origin/openshift_apiserver.go

```diff
@@ -184,26 +185,89 @@ func (c *OpenshiftAPIConfig) SkipComplete() completedConfig {
 	return completedConfig{c}
 }
 
-func (c completedConfig) newTemplateConfig() *templateapiserver.TemplateConfig {
-	return &templateapiserver.TemplateConfig{
+// legacyStorageMutator mutates the arg to modify the RESTStorage map for legacy resources
+type legacyStorageMutator interface {
+	mutate(map[schema.GroupVersion]map[string]rest.Storage)
+}
+
+type legacyStorageMutators []legacyStorageMutator
+
+func (l legacyStorageMutators) mutate(legacyStorage map[schema.GroupVersion]map[string]rest.Storage) {
+	for _, curr := range l {
+		curr.mutate(legacyStorage)
+	}
+}
+
+type legacyStorageVersionMutator struct {
```

Comment:
Even after repeated readings it's still not entirely clear to me what problem this solves... maybe add some documentation assuming very little prior knowledge?